src/Jackalope/ObjectManager.php

```diff
@@ -818,12 +818,12 @@ public function save()
             $this->executeOperations($this->operationsLog);
 
             // loop through cached nodes and commit all dirty and set them to clean.
-            if (isset($this->objectsByPath['Node'])) {
-                foreach ($this->objectsByPath['Node'] as $node) {
+            foreach ($this->objectsByPath as $nodes) {
```

Comment:
are you aware that the first level of index is the type of nodes? (e.g. with jackrabbit, there is also a VersionNode subpart in objectsByPath)

other type of objects than 'Node' should not be handled in this loop. i guess this whole object store is something we really want to clean up in 2.0 /cc @dantleech